src/dc_chat.c

```diff
@@ -2102,10 +2104,49 @@ static uint32_t send_msg_raw(dc_context_t* context, dc_chat_t* chat, const dc_ms
 	}
 	dc_param_set(msg->param, DC_PARAM_ERRONEOUS_E2EE, NULL); /* reset eg. on forwarding */
 
+	// setup In-Reply-To: and corresponding fields
+	// according to RFC 5322 3.6.4, page 25
+	stmt = dc_sqlite3_prepare(context->sql,
+		"SELECT rfc724_mid, mime_in_reply_to, mime_references"
+		" FROM msgs"
+		" WHERE timestamp=(SELECT max(timestamp) FROM msgs WHERE chat_id=? AND from_id!=?);");
+	sqlite3_bind_int  (stmt, 1, chat->id);
+	sqlite3_bind_int  (stmt, 2, DC_CONTACT_ID_SELF);
+	if (sqlite3_step(stmt)) {
+		const char* parent_rfc724_mid  = (const char*)sqlite3_column_text(stmt, 0);
+		const char* parent_in_reply_to = (const char*)sqlite3_column_text(stmt, 1);
+		const char* parent_references  = (const char*)sqlite3_column_text(stmt, 2);
+
+		if (parent_rfc724_mid && parent_rfc724_mid[0]) {
+			new_in_reply_to = dc_strdup(parent_rfc724_mid);
+		}
+
+		if (parent_references && parent_references[0]
+		 && parent_rfc724_mid && parent_rfc724_mid[0]) {
+			// angle brackets are added by the mimefactory later
+			new_references = dc_mprintf("%s %s", parent_references, parent_rfc724_mid);
+		}
+		else if (parent_references && parent_references[0]) {
+			new_references = dc_strdup(parent_references);
+		}
+		else if (parent_in_reply_to && parent_in_reply_to[0]
+		      && parent_rfc724_mid && parent_rfc724_mid[0]) {
+			new_references = dc_mprintf("%s %s", parent_in_reply_to, parent_rfc724_mid);
+		}
+		else if (parent_in_reply_to && parent_in_reply_to[0]) {
+			new_references = dc_strdup(parent_in_reply_to);
+		}
+	}
```

Comment:
i wonder what we need the references now for -- could you add a comment in the code on why this is needed?

Reply:
a few lines above there was an, indeed, very short hint to the corresponding RFC; i added some more lines in  https://github.com/deltachat/deltachat-core/pull/407/commits/c7a4e76b09df261ae439f2aacac8aef6c0f6f071